src/useApiData.test.ts

```diff
@@ -57,11 +57,25 @@ describe('useApiData should trigger perform under the right conditions', () => {
             initialProps: { endpointKey: 'testEndpoint', params: {} },
         });
         expect(performMock1).toHaveBeenCalledTimes(1);
-        rerender();
+        expect(performMock1).toHaveBeenCalled();
+        // set ready to success to make a new store to test the invalidations.
```

Comment:
Ik snap niet echt wat je hier hebt gedaan. Ik dacht dat je een test zou toevoegen bij de param change, dat dit wel retriggert bij het veranderen van param values, maar niet bij veranderen van params object met dezelfde values.

Reply:
Een nieuwe leeg object is ook een nieuw object met met dezelfde values. Dat test ik nu in deze use case. Probleem met hoe de test eerst was is dat ik hem alleen rerenderde en niet een nieuwe object meegaf aan de hook. Daarnaast zette ik de networkStatus niet daadwerkelijk op failed(dit is een enorm gedoe) . Voor mutatie van de params is al een andere test.